spec/dataframe_spec.rb

```diff
@@ -3050,10 +3050,24 @@
   end
 
   context "#summary" do
-    it "produces a summary of data frame" do
+    it "shows the dataframe name in summary" do
       expect(@data_frame.summary.match("#{@data_frame.name}")).to_not eq(nil)
       expect(@df_mi.summary.match("#{@df_mi.name}")).to_not eq(nil)
     end
+
+    it "shows the no of rows in dataframe" do
+      expect(@data_frame.summary.match("Number of rows: #{@data_frame.nrows}")).to_not eq(nil)
+      expect(@df_mi.summary.match("Number of rows: #{@df_mi.nrows}")).to_not eq(nil)
+    end
+
+    it "shows the individual vector names in the dataframe" do
+      expect(@data_frame.summary.match(/Element:\[#{@data_frame.numeric_vector_names[0]}\]/)).to_not eq(nil)
+    end
+
+    it "shows the summary of individual vectors in dataframe" do
+      expect(@data_frame.summary.match(@data_frame[:a].summary(1))).to_not eq(nil)
+      expect(@df_mi.summary.match(@df_mi[[:a, :one, :bar]].summary(1))).to_not eq(nil)
+    end
```

Comment:
To be honest, I feel like those "match this"/"match that" tests are fragile and not very informative (you can't understand from tests what exactly summary outputs). Please take a look at how `#inspect` is tested: https://github.com/SciRuby/daru/blob/master/spec/dataframe_spec.rb#L3648